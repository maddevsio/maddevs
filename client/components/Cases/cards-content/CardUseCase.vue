<template>
  <div class="card-use-case">
    <client-only>
      <h4 class="card-use-case__title case_title_h4" :class="`card-use-case__${classList.uniqueСlass}-title`" v-html="title">
        {{title}}
      </h4>
    </client-only>
    <div class="card-use-case__description-wrapper" :class="`card-use-case__${classList.uniqueСlass}-description-wrapper`">
      <TextParagraph class="card-use-case__paragraph" :class="`card-use-case__${classList.uniqueСlass}-paragraph`">
        {{description}}
      </TextParagraph>
      <div :class="`card-use-case__${classList.uniqueСlass}-picture`" class="card-use-case__desktop-picture">
        <Picture
          :pictureFolder="pictureFolder"
          :fileName="fileName"
          :fileExtension="fileExtension"
          :alt="alt"
          :lazyLoading="false"
        />
      </div>
      <div class="card-use-case__mobile-picture" :class="`card-use-case__${classList.uniqueСlass}-mobile-picture`">
        <Picture
          :pictureFolder="pictureFolder"
          :fileName="`${fileName}-mobile`"
          :fileExtension="fileExtension"
          :alt="alt"
          :lazyLoading="false"
        />
      </div>
    </div>
  </div>
</template>

<script>
import TextParagraph from '@/components/Cases/TextParagraph';
import Picture from '@/components/Cases/Picture';
export default {
  name: 'CardUseCase',
  components: {
    TextParagraph,
    Picture
  },
  props: {
    title: {
      type: String,
      default: ''
    },
    description: {
      type: String,
      default: ''
    },
    pictureFolder: {
      type: String,
      default: ''
    },
    fileName: {
      type: String,
      default: ''
    },
    fileExtension: {
      type: String,
      default: ''
    },
    alt: {
      type: String,
      default: ''
    },
    classList: {
      type: Object
    }
  }
};
</script>

<style scoped lang="scss">
  @import '../../../assets/styles/cases/_mixins';

  .card-use-case {
    height: 100%;
    display: flex;
    flex-direction: column;
    position: relative;

    &__description-wrapper {
      display: grid;
      grid-template-columns: repeat(2, 1fr);
      height: 100%;
      margin-bottom: -32px;
    }

    &__title {
      margin-bottom: 11px;
    }

    &__title,
    &__paragraph {
      color: $text-color--white-primary;
    }

    &__white-card-title,
    &__white-card-paragraph {
      color: $text-color--black-cases;
    }

    &__desktop-picture {
      margin-top: auto;
      position: relative;
      right: -32px;
    }

    &__red-card-picture,
    &__black-card-picture {
      width: 140px;
    }

    &__turquoise-card-picture,
    &__white-card-picture {
      width: 166px;
    }

    &__mobile-picture {
      display: none;
    }

    &__turquoise-card-paragraph {
      padding-bottom: 40px;
    }

    /deep/ span {
      white-space: nowrap;
    }

    @media screen and (max-width: 1000px) {
      text-align: center;

      &__title {
        /deep/ br {
          display: none;
        }
      }

      &__turquoise-card-paragraph {
        padding-bottom: 0;
      }

      &__paragraph {
        margin-bottom: 24px;
      }

      &__desktop-picture {
        display: none;
      }

      &__mobile-picture {
        display: block;
        width: 100%;
        height: auto;
      }

      &__description-wrapper {
        height: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        align-items: center;
        margin-bottom: -33px;
      }

      &__black-card-description-wrapper {
        margin-bottom: -7.1vw;
      }

      &__white-card-mobile-picture {
        margin-bottom: -6.4vw;
      }
    }

    @media screen and (max-width: 768px) {
      &__description-wrapper {
        margin-bottom: -21px;
      }

      &__black-card-description-wrapper {
        margin-bottom: -6.6vw;
      }
    }

    @media screen and (max-width: 898px) {
      &__white-card-mobile-picture {
        margin-bottom: -5.9vw;
      }
    }

    @media screen and (max-width: 620px) {
      &__mobile-picture {
        max-width: 320px;
      }

      &__black-card-description-wrapper {
        margin-bottom: -21px;
      }

      &__white-card-mobile-picture {
        margin-bottom: 0;
      }
    }
  }
</style>
