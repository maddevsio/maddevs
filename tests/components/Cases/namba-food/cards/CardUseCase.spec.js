import CardUseCase from '@/components/Cases/nambafood/cards/CardUseCase.vue'
import { render, screen } from '@testing-library/vue'

const props = {
  title: 'some-title',
  description: 'some-description',
  pictureFolder: '',
  fileName: 'some-filename',
  fileExtension: 'png',
  alt: 'alt',
}

describe('CardUseCase component', () => {
  it('should render correctly', () => {
    const { container } = render(CardUseCase, {
      props,
      stubs: ['ClientOnly'],
      mocks: {
        getMediaFromS3: () => 'img.jpg',
      },
    })

    expect(container).toMatchSnapshot()
  })

  it('should correctly render classname', () => {
    const classList = {
      uniqueСlass: 'uniqueСlass',
    }
    const { html } = render(CardUseCase, {
      props: {
        ...props,
        classList,
      },
      stubs: ['ClientOnly'],
      mocks: {
        getMediaFromS3: () => 'img.jpg',
      },
    })

    expect(screen.getByText(props.title)).toBeTruthy()
    expect(html()).toContain(`card-use-case__${classList.uniqueСlass}-title`)
  })
})
