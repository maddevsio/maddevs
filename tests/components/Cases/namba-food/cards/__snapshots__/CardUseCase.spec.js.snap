// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardUseCase component should render correctly 1`] = `
<div>
  <div
    class="card-use-case"
  >
    <clientonly-stub>
      <h3
        class="card-use-case__title case_title_h4 card-use-case__undefined-title"
      >
        some-title
      </h3>
    </clientonly-stub>
     
    <div
      class="card-use-case__description-wrapper card-use-case__undefined-description-wrapper"
    >
      <p
        class="case_paragraph card-use-case__paragraph card-use-case__undefined-paragraph"
      >
        
      some-description
    
      </p>
       
      <div
        class="card-use-case__desktop-picture card-use-case__undefined-picture"
      >
        <picture>
          <source
            class="image"
            srcset="img.jpg ,img.jpg 2x"
            type="image/webp"
          />
           
          <img
            alt="alt"
            class="image"
            data-testid="test-picture-img"
            height="0"
            loading=""
            src="img.jpg"
            srcset="img.jpg 2x"
            width="0"
          />
        </picture>
      </div>
       
      <div
        class="card-use-case__mobile-picture card-use-case__undefined-mobile-picture"
      >
        <picture>
          <source
            class="image"
            srcset="img.jpg ,img.jpg 2x"
            type="image/webp"
          />
           
          <img
            alt="alt"
            class="image"
            data-testid="test-picture-img"
            height="0"
            loading=""
            src="img.jpg"
            srcset="img.jpg 2x"
            width="0"
          />
        </picture>
      </div>
    </div>
  </div>
</div>
`;
